lib/model/QubitRepository.php

```diff
@@ -124,24 +124,45 @@ public function save($connection = null)
     return $this;
   }
 
+  public function getRelatedInformationObjectIds()
+  {
+    $sql = "SELECT id FROM ". QubitInformationObject::TABLE_NAME ." WHERE repository_id=:repository_id";
+
+    $params = array(':repository_id' => $this->id);
+    $results = QubitPdo::fetchAll($sql, $params, array('fetchMode' => PDO::FETCH_ASSOC));
+
+    $ioIds = array();
+
+    foreach ($results as $row)
+    {
+      array_push($ioIds, $row['id']);
+    }
+
+    return $ioIds;
+  }
+
   public function updateRelatedIos()
   {
     $ioIds = array();
     $context = sfContext::getInstance();
     $env = $context->getConfiguration()->getEnvironment();
 
-    foreach ($this->informationObjects as $io)
+    foreach ($this->getRelatedInformationObjectIds() as $id)
     {
       // Update synchronously in CLI tasks and jobs
       if (in_array($env, array('cli', 'worker')))
       {
+        $io = QubitInformationObject::getById($id);
         QubitSearch::getInstance()->update($io, array('updateDescendants' => true));
```

Comment:
Could [arElasticSearchPlugin::partialUpdate()](https://github.com/artefactual/atom/blob/qa/2.6.x/plugins/arElasticSearchPlugin/lib/arElasticSearchPlugin.class.php#L473) be used in this context? It may be more efficient then a full update if it's updating a small number of fields.

Reply:
That's a great point @djjuhasz. At the moment the partial update function doesn't consider descendants and that's required here. I believe this is kind of urgent, so maybe that fits better on the work you started to reduce the amount of data from related resources in the IO documents. I hope we can improve that soon in the current projects going on.

Reply:
Good call. I'm going to make a separate issue for that, though, given it'll take more time to test.